app/models/publication.rb

```diff
@@ -590,6 +606,10 @@ def update_from_unpaywall(unpaywall_response)
     end
   end
 
+  def update_oa_status_from_unpaywall(unpaywall_response)
+    self.open_access_status = doi.present? || unpaywall_response.matchable_title == matchable_title ? unpaywall_response.oa_status : 'unknown'
```

Comment:
I think we should add a unit test for this.  The order of operations here makes me a bit uneasy, but I suspect this works.